cloudsdk/src/main/java/io/particle/android/sdk/cloud/ParticleDevice.java

```diff
@@ -471,20 +479,105 @@ private void resetFlashingState() {
     private void performFlashingChange(FlashingChange flashingChange) throws ParticleCloudException {
         try {
             flashingChange.executeFlashingChange();
-            isFlashing = true;
-            cloud.notifyDeviceChanged();
-            // Gross.  We're using this "run delayed" hack just for the *scheduling* aspect
-            // of this, and then we're just telling the scheduled runnable to drop right back to a
-            // background thread so we don't call resetFlashingState() on the main thread.
-            // Still, I don't want to introduce a whole scheduled executor setup *just for this*,
-            // or write something that just sits in a Thread.sleep(), hogging a whole thread when
-            // more important work could be getting blocked.
-            EZ.runOnMainThreadDelayed(30000, () -> EZ.runAsync(() -> resetFlashingState()));
-        } catch (RetrofitError e) {
+            //listens for flashing event, on success unsubscribe from listening.
+            subscribeToSystemEvent("spark/flash/status", new SimpleParticleEventHandler() {
+                @Override
+                public void onEvent(String eventName, ParticleEvent particleEvent) {
+                    if (particleEvent.dataPayload.equals("success")) {
```

Comment:
might want to do this the other way 'round, given how many things are permitted be null. :(  `"success".equals(...)`